src/components/MainComposer/components/MainComposerBase/MainComposerBase.tsx

```diff
@@ -1,42 +1,62 @@
-import { ReactNode } from 'react';
+import { ReactNode, useState } from 'react';
+
+import { useError } from '~stores/useError/useError';
 
 import ComposerEditor from '../ComposerEditor/ComposerEditor';
 import InputMediaGroup from '../InputMediaGroup/InputMediaGroup';
 
 import scss from './MainComposerBase.module.scss';
 
-import { ErrorMapText } from '../ComposerEditor/ComposerEditor.types';
-import { ErrorMediaInput } from '../InputMediaGroup/InputMediaGroup.type';
-import { MainComposerBaseProps } from './MainComposerBase.type';
+import { Error, MainComposerBaseProps } from './MainComposerBase.type';
 
 function MainComposerBase(props: MainComposerBaseProps): ReactNode {
-  const handleMediaErrors = (errors: ErrorMediaInput): void => {
-    props.onErrorMedia?.(errors);
+  const [errors, setErrors] = useState<Record<string, string>>({});
+
+  const errorStore = useError();
+
+  const addErrors = (id: string, error: Error): void => {
+    const { message } = error;
+    const useErrorId = errorStore.addError({ message });
+
+    setErrors({ ...errors, [id]: useErrorId });
+    props.onError?.(true);
+  };
+
+  const errorRemover = (id: string): void => {
+    if (id) {
+      props.onError?.(false);
+      errorStore.removeError(errors[id]);
+    }
   };
 
-  const handleTextErrors = (errors: ErrorMapText): void => {
-    props.onErrorText?.(errors);
+  const renderImputMediaGroup = (): ReactNode => {
+    if (props.postMode && 'media' in props.postMode.validators) {
+      return (
+        <InputMediaGroup
+          accountId={props.accountId}
+          addError={addErrors}
+          postMode={props.postMode}
+          removeError={errorRemover}
+        />
+      );
+    }
   };
 
   return (
     <div className={scss.container}>
       <ComposerEditor
         accountId={props.accountId}
+        addError={addErrors}
         currentMaxLimit={props.currentMaxLimit ?? undefined}
-        onChangePost={props.onChange}
-        onError={handleTextErrors}
+        onChange={props.onChange}
         postMode={props.postMode}
+        removeError={errorRemover}
         value={props.value}
       />
       <div className={scss.bottomWrapper}>
         <hr className={scss.divider} />
-        <div>
-          <InputMediaGroup
-            accountId={props.accountId}
-            onError={handleMediaErrors}
-            postMode={props.postMode}
-          />
-        </div>
+        {props.postMode ? renderImputMediaGroup() : <InputMediaGroup />}
```

Comment:
isso acabou não ficando claro, se props.postMode for verdadeiro, ele renderiza InputMediaGroup, se não, ele renderiza o componente InputMediaGroup. Parece a mesma coisa